pkg/providers/vsphere/validator.go

```diff
@@ -580,29 +587,46 @@ func (v *Validator) validateCPUserPrivs(ctx context.Context, spec *Spec, vuc *co
 		datacenter,
 	)
 	if err != nil {
-		return err
+		return false, err
 	}
 
 	return v.validatePrivs(ctx, privObjs, vsc)
 }
 
-func (v *Validator) validatePrivs(ctx context.Context, privObjs []PrivAssociation, vsc govmomi.VSphereClient) error {
-	var missingPrivs []string
+func (v *Validator) validatePrivs(ctx context.Context, privObjs []PrivAssociation, vsc govmomi.VSphereClient) (bool, error) {
+	var privs []string
 	var err error
+	missingPrivs := []missingPriv{}
+	passed := false
 
 	for _, obj := range privObjs {
 		path := obj.path
 		privsContent := obj.privsContent
 		t := obj.objectType
-		missingPrivs, err = v.getMissingPrivs(ctx, vsc, path, t, privsContent, vsc.Username())
+		privs, err = v.getMissingPrivs(ctx, vsc, path, t, privsContent, vsc.Username())
 		if err != nil {
-			return err
-		} else if len(missingPrivs) > 0 {
-			return fmt.Errorf("User %s missing privileges on %s: %v", vsc.Username(), path, missingPrivs)
+			return passed, err
+		} else if len(privs) > 0 {
+			missingPrivs = append(missingPrivs, missingPriv{
+				username:   vsc.Username(),
+				objectType: t,
+				path:       path,
+				privs:      privs,
+			})
 		}
 	}
 
-	return nil
+	if len(missingPrivs) == 0 {
+		passed = true
+	} else {
+		content, err := json.MarshalIndent(missingPrivs, " ", " ")
+		if err != nil {
```

Comment:
What is the situation that it would go here?

Reply:
None that I can think of, and it's impossible to test 😂 I thought I remembered the linter complaining about unchecked error values. Maybe I made that up.